### PR TITLE
ingress-gateway: add initial demo files for ingress gateways

### PR DESCRIPTION
**Pull request by shoenig**

### CODE REVIEW

File: ingress-gateway/ig-bridge-demo.nomad [hunk 1]
```diff
@@ -0,0 +1,96 @@
+job "ig-bridge-demo" {
+
+  datacenters = ["dc1"]
+
+  group "ingress-group" {
```
Comment (nickethier) [inline ingress-gateway/ig-bridge-demo.nomad:5]: I suggest a comment here explaining why this group has no task blocks.

File: ingress-gateway/ig-bridge-demo.nomad [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    service {
+      name = "my-ingress-service"
+      port = "8080"
+
+      connect {
+        gateway {
+
+          // Consul gateway [envoy] proxy options.
```
Comment (nickethier) [inline ingress-gateway/ig-bridge-demo.nomad:25]: nit: can we not mix comment styles :grimacing:

File: ingress-gateway/ig-demo.nomad [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  # The UUID generator from the connect-native demo is used as an example service.
+  # The ingress gateway above makes access to the service possible over normal HTTP.
+  # For example,
+  #
+  # $ curl $(dig +short @127.0.0.1 -p 8600 uuid-api.ingress.dc1.consul. ANY):8080
+  group "generator" {
+    network {
+      mode = "host"
+      port "api" {
+        to = -1
```
Comment (nickethier) [inline ingress-gateway/ig-demo.nomad:70]: do you need this if network mode is "host"?

**Comment by nickethier** [comment]
Only blocker was my question around the port block

File: ingress-gateway/ig-demo.nomad (hunk 3 above)
Comment (shoenig) [reply ingress-gateway/ig-demo.nomad:70]: Agh good catch! Old habits die hard..

**Pull request merged**